File: 0049-group-anagrams/0049-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
   let result = {}
   for (let str of strs) {
       let key = str.split('').sort().join('')
       if (key in result) result[key].push(str)
       else result[key] = [str]
   }
   return Object.values(result)
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        storage = {}
        for str in strs:
            sortedStr = [x for x in str]
            sortedStr.sort()
            sortedStr = "".join(sortedStr)
            if sortedStr in storage:
                storage[sortedStr].append(str)
            else:
                storage[sortedStr] = [str]
        return storage.values()
        